src/main/kotlin/org/jetbrains/research/testspark/tools/ToolUtils.kt

```diff
@@ -61,27 +62,29 @@ fun saveData(
     report: Report,
     packageLine: String,
     importsCode: MutableSet<String>,
+    fileUrl: String,
+    generatedTestData: TestGenerationData,
 ) {
-    project.service<TestGenerationDataService>().resultName = project.service<TestStorageProcessingService>().testResultName
-    project.service<TestGenerationDataService>().fileUrl = project.service<ProjectContextService>().fileUrl!!
-    project.service<TestGenerationDataService>().packageLine = packageLine
-    project.service<TestGenerationDataService>().importsCode.addAll(importsCode)
+    generatedTestData.fileUrl = fileUrl
+    generatedTestData.packageLine = packageLine
+    generatedTestData.importsCode.addAll(importsCode)
```

Comment:
*Just some thoughts, not a comment.
In the future it would be very cool to make data storage classes such as `TestGenerationData` and others be immutable. Because having mutable fields of some of them makes it hard to understand/keep in mind where these mutable fields are set/changed.